posts/coder-and-nix/index.md

```diff
@@ -0,0 +1,117 @@
+
+The nix-shell command resolves many of the uncertainties of the shared development process. You can use nix-shell from the command line by using the nix-shell command directly to install a package. For example, if you use the [fzf](https://github.com/junegunn/fzf) tool as a part of your workflow you can install it using the following command:
+
+```bash
+nix-shell -p fzf
+
+```
+
+The above command will add `fzf` to your local nix store and temporarily modify your $PATH variable to include the command in your shell. `fzf` will be available immediately. 
+
+On the other hand, if you're getting started with nix and want to create an environment that contains many of your favorite tools you can create a `shell.nix` file. This file will contain the commands needed to make the tools available for your current session, and requires the Nix builtin function (nix calls these functions [derivations](https://nixos.org/manual/nix/stable/language/derivations.html)) `mkshell`. The following command will install `zsh`, `fzf`, `autoenv`, and the `powerlevel10k` theme.
+
+```nix
+# shell.nix
+
+with (import <nixpkgs> {});
+mkshell {
+  buildInputs = [
+    fzf
+    zsh
+    zsh-autoenv
+    zsh-powerlevel10k
+  ];
+}
+```
+
+To use this `shell.nix` file you can run `nix-shell` without any arguments. The `nix-shell` tool will look for this file by default. 
+
+### Adding Your Tools to A Shared `shell.nix` Configuration
+
+The main benefit over using Docker for shared development environments is that Nix has the ability to extend this shared configuration with your own personal settings. For example, if you want to add your toolchain to a [nodejs.nix development environment](https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/web/nodejs/nodejs.nix), you can modify your `shell.nix` file with the following code. Note that the `nodejs.nix` file is imported as part of the `buildInputs` array.
+
+```nix
+# shell.nix
+
+with (import <nixpkgs> {});
+mkshell {
+  buildInputs = [
+    (import ./nodejs.nix { inherit pkgs; })
+    fzf
+    zsh
+    zsh-autoenv
+    zsh-powerlevel10k
+  ];
+}
+```
+
+Now running the `nix-shell` command will create an environment containing the shared NodeJS nix configuration and your custom toolchain. 
+
+## Using Nix with Coder
+
+Now that you have your NodeJS environment and custom toolchain setup in `shell.nix`, lets add Coder to your environment. Coder is available as a nix package and can be added to your config as the same way you have added your custom software in your tool chain. You can do so by modifying your `shell.nix` file to include the Coder package, like in the example code below:
```

Comment:
Let's take another direction with this section. Instead of showing how to install the Coder CLI locally, let's provide an example template in Coder that users can choose to create remote workspaces that contain the `nix` shell. These workspaces could be provisioned on AWS, Docker, or Kubernetes. I have no preference, but we could show each 🤷🏼

Then, each developer could spin up a workspace and use `nix` to get the tools they need for their workspace inside the home directory. If it'd be helpful, I can set you up with a Coder workspace and we can work through this template.

Reply:
Ok, that works for me. I initially had a section discussing templates but I wasn't sure how to connect it to the rest of the article. 

It would be helpful to work through the template together so I can get an idea of how it all works. Thanks!